vllm/model_executor/model_loader/loader.py

```diff
@@ -44,6 +44,74 @@
 from vllm.platforms import current_platform
 from vllm.utils import is_pin_memory_available
 
+# _NN_MODULE_TO_COMPILE[torch.nn.Linear] = torch.compile means
+# that after torch.nn.Linear is loaded, it will be compiled using torch.compile
+# Search for _NN_MODULE_TO_COMPILE in the codebase to see how it is used.
+_NN_MODULE_TO_COMPILE: Dict[torch.nn.Module, Callable[[torch.nn.Module],
+                                                      Callable], ] = {}
+
+
+def register_module_to_compile(
+    module: torch.nn.Module,
+    compiler: Callable[[torch.nn.Module], Callable],
+) -> None:
+    """
+    Registers a neural network module with a specified compiler function.
+
+    This function ensures that the given module is not already registered
```

Comment:
It allows repeated registering for the same nn.Module. It's possible that the both llama and phi want to compile the same sub-modules. On the other hand, we can create model-specific compiler pool (i.e., phi-3-small and llama-3.1 have their own _NN_MODULE_TO_COMPILE. Thoughts?